storage_service/locations/models/wellcome.py

```diff
@@ -253,30 +154,31 @@ def move_from_storage_service(self, src_path, dest_path, package=None):
             with open(src_path, 'rb') as data:
                 bucket.upload_fileobj(data, s3_path)
 
-            wellcome = WellcomeStorageServiceClient(self.api_root_url, {
-                'token_url': self.token_url,
-                'client_id': self.app_client_id,
-                'client_secret': self.app_client_secret,
-            })
+            wellcome = StorageServiceClient(
+                api_url=self.api_root_url,
+                token_url=self.token_url,
+                client_id=self.app_client_id,
+                client_secret=self.app_client_secret,
+            )
 
             callback_url = urljoin(
                 self.callback_host,
-                '/api/v2/file/%s/wellcome_callback/?%s' % (
-                package.uuid,
-                urlencode({
-                    'username': self.callback_username,
-                    'api_key': self.callback_api_key,
-                })))
+                '%s?%s' % (
+                    reverse('wellcome_callback', args=['v2', 'file', package.uuid]),
+                    urlencode({
+                        'username': self.callback_username,
+                        'api_key': self.callback_api_key,
```

Comment:
I’m a bit dubious about sending URLs with API keys into the storage service, because we don’t have any encryption on that data at rest.

It doesn’t need to be changed in this patch, but I wonder if we can do without them?

Reply:
I agree. Would need the storage service to support sending an auth token header when it does the callback.